apps/system/lockscreen/js/lockscreen_state_manager.js

```diff
@@ -82,6 +82,7 @@
         'screenchange',
         'lockscreen-notification-request-activate-unlock',
         'lockscreen-request-unlock',
+        'lockscreen-appopened',
```

Comment:
It's better to add a workaround comment:

    // XXX: this is necessary only when LockScreen is not an iframe.
    // Since for an iframe it will not listen to any event after unlocking, namely when it
    // is pushed into the background. Nevertheless we need to do the same suspending/resuming
    // thing in this way.

You can find a suitable place to put the sentences like that. It elaborates the motivation why we need to do that now.